rclcpp_lifecycle/test/benchmark/benchmark_lifecycle_client.cpp

```diff
@@ -0,0 +1,307 @@
+    }
+
+    if (!future_result.valid()) {
+      throw std::runtime_error("Future result was not valid");
+    }
+
+    return future_result.get()->available_states;
+  }
+
+  std::vector<lifecycle_msgs::msg::TransitionDescription>
+  get_available_transitions(std::chrono::seconds time_out = 1s)
+  {
+    auto request = std::make_shared<lifecycle_msgs::srv::GetAvailableTransitions::Request>();
+
+    if (!client_get_available_transitions_->wait_for_service(time_out)) {
+      throw std::runtime_error("Wait for service timed out");
+    }
+
+    auto future_result = client_get_available_transitions_->async_send_request(request);
+    auto future_status = future_result.wait_for(time_out);
+
+    if (future_status != std::future_status::ready) {
+      throw std::runtime_error("Get available transitions request failed");
+    }
+
+    if (!future_result.valid()) {
+      throw std::runtime_error("Future result was not valid");
+    }
+
+    return future_result.get()->available_transitions;
+  }
+
+  std::vector<lifecycle_msgs::msg::TransitionDescription>
+  get_transition_graph(std::chrono::seconds time_out = 1s)
+  {
+    auto request = std::make_shared<lifecycle_msgs::srv::GetAvailableTransitions::Request>();
+
+    if (!client_get_transition_graph_->wait_for_service(time_out)) {
+      throw std::runtime_error("Wait for service timed out");
+    }
+
+    auto future_result = client_get_transition_graph_->async_send_request(request);
+    auto future_status = future_result.wait_for(time_out);
+
+    if (future_status != std::future_status::ready) {
+      throw std::runtime_error("Get transition graph request failed");
+    }
+
+    if (!future_result.valid()) {
+      throw std::runtime_error("Future result was not valid");
+    }
+
+    return future_result.get()->available_transitions;
+  }
+
+private:
+  std::shared_ptr<rclcpp::Client<lifecycle_msgs::srv::GetAvailableStates>>
+  client_get_available_states_;
+  std::shared_ptr<rclcpp::Client<lifecycle_msgs::srv::GetAvailableTransitions>>
+  client_get_available_transitions_;
+  std::shared_ptr<rclcpp::Client<lifecycle_msgs::srv::GetAvailableTransitions>>
+  client_get_transition_graph_;
+  std::shared_ptr<rclcpp::Client<lifecycle_msgs::srv::GetState>> client_get_state_;
+  std::shared_ptr<rclcpp::Client<lifecycle_msgs::srv::ChangeState>> client_change_state_;
+};
+
+class BenchmarkLifecycleClient : public performance_test_fixture::PerformanceTest
+{
+public:
+  void SetUp(benchmark::State & state)
+  {
+    rclcpp::init(0, nullptr);
+    lifecycle_node = std::make_shared<rclcpp_lifecycle::LifecycleNode>(lifecycle_node_name);
+    lifecycle_client = std::make_shared<LifecycleServiceClient>("lifecycle_client");
+    executor = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+    executor->add_node(lifecycle_node->get_node_base_interface());
+    executor->add_node(lifecycle_client->get_node_base_interface());
+    spinner_ = std::thread([this]() {executor->spin();});
```

Comment:
There's really no need for a lambda here.
```suggestion
    spinner_ = std::thread(&rclcpp::executors::SingleThreadedExecutor::spin, executor);
```

Reply:
Replaced